From 4226bbf4267a7dd88dc2fab1c84a4c95cb3e6314 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Tue, 22 Nov 2022 22:08:59 +0000
Subject: [PATCH 3/5] gnu: Add cl-assert-p.

* gnu/packages/lisp-check.scm (cl-assert-p, ecl-assert-p,
  sbcl-assert-p): New variables.
---
 gnu/packages/lisp-check.scm | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/gnu/packages/lisp-check.scm b/gnu/packages/lisp-check.scm
index 2d447533d9..9b22b195b5 100644
--- a/gnu/packages/lisp-check.scm
+++ b/gnu/packages/lisp-check.scm
@@ -123,6 +123,32 @@ (define-public ecl-assertion-error
 (define-public cl-assertion-error
   (sbcl-package->cl-source-package sbcl-assertion-error))
 
+(define-public sbcl-assert-p
+  (package
+    (name "sbcl-assert-p")
+    (version "1.0.1")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/noloop/assert-p")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name "assert-p" version))
+       (sha256
+        (base32 "1x24rkqkqiw8zd26swi9rmhfplkmr5scz3bhjwccah9d2s36b1xs"))))
+    (build-system asdf-build-system/sbcl)
+    (inputs (list sbcl-assertion-error sbcl-simplet))
+    (home-page "https://github.com/noloop/assert-p")
+    (synopsis "Common Lisp assertion library")
+    (description "This package provides a Common Lisp collection of assertions.")
+    (license license:gpl3)))
+
+(define-public ecl-assert-p
+  (sbcl-package->ecl-package sbcl-assert-p))
+
+(define-public cl-assert-p
+  (sbcl-package->cl-source-package sbcl-assert-p))
+
 (define-public sbcl-check-it
   (let ((commit "b79c9103665be3976915b56b570038f03486e62f"))
     (package
-- 
2.38.0

